### ISSUE TITLE
`~/Library/Preferences` should not be used on macOS

### ISSUE THREAD

**Issue by kidonng** [muesli/go-app-paths#6]
This has been discussed in other places, like dirs-dev/directories-rs#62.
The basic idea is that this folder should only be interacted via macOS's native API, otherwise apps should use `~/Library/Application Support`.

**Comment by muesli** [comment]
Obviously not easy to solve without breaking API compatibility. Are there any actual real-life issues with applications using this path for non-plist files?

**Comment by kidonng** [comment]
> Obviously not easy to solve without breaking API compatibility.

I agree this is tough. Migrating settings is always challenging, but the benefits of a better default certainly precedes that.

> Are there any actual real-life issues with applications using this path for non-plist files?

As far as I know, not really. Even some prominent macOS developers (e.g. Parallels) and frameworks (e.g. QT, though I'm not sure) does this.

But as mentioned in the referenced issue, that's still against Apple's documentation. This issue is to raise concern so we can avoid potential issues in the future.

**Comment by deanishe** [comment]
> Are there any actual real-life issues with applications using this path for non-plist files?

Users and software trying to uninstall the offending application may fail to find the files because they're in a place they shouldn't be.

Also, macOS sandboxing gets more restrictive with each release, and I don't think it would surprise anybody if Apple did lock down the folder at some point. They do not care in the slightest about breaking applications that aren't following the rules.

Shame the docs weren't consulted before implementing this.

**Comment by batmac** [comment]
Hi,
I understand that this is very hard to change because of existing apps, but what approach do you think I could take for a new app, because I would like to use the lib but writing to "~/Library/Preferences" is roughly a no-go ? Is there for instance a way to get XDG dirs on MacOS?

**Comment by kidonng** [comment]
"no-go" is a little bit too much, I actually don't see any problem using `Preferences` in the foreseeable future, if we are being practical. It's just inappropriate, but so widely misused it sort of become a semi-standard.

There's basically no XDG stuff on macOS. Everything similar to that is just convention. That said, many programs do still respect `$XDG_*` environment variables. So if you feel like `Application Support` is not your thing, `~/.config` is super fine (and in fact the most used location by CLI programs).

**Comment by batmac** [comment]
thank you for your help !
